### PR TITLE
Update TF registry docs to have user_labels nested under settings and…

### PR DESCRIPTION
**Pull request by AmandaHassoun**
… not settings.advanced_machine_features

I was getting the following failure originally:
```
╷
│ Error: Unsupported argument
│
│   on modules/grafana/sql.tf line 30, in resource "google_sql_database_instance" "instance":
│   30:       user_labels = {
│
│ An argument named "user_labels" is not expected here.
```

Until I looked through the JSON API spec and realized it's supposed to only be nested under settings (same level as tier).

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [x] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read [Write release notes](https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none
```

### CODE REVIEW

**Comment by AmandaHassoun** [comment]
I signed the CLA but unsure how to re-trigger that job 😓

**Comment by c2thorn** [comment]
> I signed the CLA but unsure how to re-trigger that job 😓

I have to give approval, should be running now. Also, I went ahead and edited the release note. Documentation-only changes don't need a note

File: mmv1/third_party/terraform/website/docs/r/sql_database_instance.html.markdown
Comment (c2thorn) [inline mmv1/third_party/terraform/website/docs/r/sql_database_instance.html.markdown:251]: `settings.advanced_machine_features` being this high is probably adding to the confusion as well. It should probably go lower, probably right above the `settings.database_flags` block. Do you mind moving that ( and it's child `threads_per_core` field as well? @AmandaHassoun

Comment (AmandaHassoun) [reply mmv1/third_party/terraform/website/docs/r/sql_database_instance.html.markdown:251]: will do @c2thorn 👍🏼

Comment (c2thorn) [reply mmv1/third_party/terraform/website/docs/r/sql_database_instance.html.markdown:251]: `settings.advanced_machine_features` only supports the `settings.advanced_machine_features.threads_per_core` field, but it makes it look like fields like `activation_policy` `availability_type`, etc are under it as well. I think that is maybe why you thought `settings.user_labels` is under `settings.advanced_machine_features`.

Line 283 is when we start listing all of the other parent fields with subfields, so we should move `settings.advanced_machine_features` somewhere down there. Right before `settings.database_flags` would be nice so that there is some alphabetical ordering.

Comment (AmandaHassoun) [reply mmv1/third_party/terraform/website/docs/r/sql_database_instance.html.markdown:251]: @c2thorn updated! Also, I have no access to the failures and they seem to be unrelated to my changes..

Comment (AmandaHassoun) [reply mmv1/third_party/terraform/website/docs/r/sql_database_instance.html.markdown]: I did totally misunderstand.. Fixed!

Comment (AmandaHassoun) [reply mmv1/third_party/terraform/website/docs/r/sql_database_instance.html.markdown]: To keep alphabetical ordering, I put `threads_per_core` before `time_zone`.

**Comment by c2thorn** [comment]
Apologies for the miscommunication, I went ahead and committed what I was trying to convey. I'll merge after our CI runs one last time!

**Review by c2thorn — APPROVED** [review]

**Pull request merged**
